spec/handlers/newflow/confirm_oauth_info_spec.rb

```diff
@@ -24,37 +21,91 @@ module Newflow
+      end
 
-    it 'changes the user state to "activated"' do
-      expect(User.last.state).to_not eq('activated')
-      described_class.call(params: params, user: User.last)
-      expect(User.last.state).to eq('activated')
-    end
+      it 'adds the user as a "lead" to salesforce' do
+        expect_any_instance_of(PushSalesforceLead).to receive(:exec)
+        described_class.call(params: params, user: User.last)
+      end
+
+      it 'changes the user state to "activated"' do
+        expect(User.last.state).to_not eq('activated')
+        described_class.call(params: params, user: User.last)
+        expect(User.last.state).to eq('activated')
+      end
 
-    it 'agrees to terms of use and privacy policy when contracts_required' do
-      # TODO: ideally would do this but it fails with an error:
-      # expect_any_instance_of(AgreeToTerms).to receive(:call).twice.and_call_original
+      it 'agrees to terms of use and privacy policy when contracts_required' do
+        # TODO: ideally would do this but it fails with an error:
+        # expect_any_instance_of(AgreeToTerms).to receive(:call).twice.and_call_original
 
-      expect {
+        expect {
+          described_class.call(params: params, contracts_required: true, user: User.last)
+        }.to change {
+          FinePrint::Signature.count
+        }.by(2)
+      end
+
+      it 'signs up user for the newsletter when checked' do
+        expect_any_instance_of(PushSalesforceLead).to receive(:exec).with(hash_including({ newsletter: true }))
         described_class.call(params: params, contracts_required: true, user: User.last)
-      }.to change {
-        FinePrint::Signature.count
-      }.by(2)
-    end
+      end
 
-    it 'signs up user for the newsletter when checked' do
-      expect_any_instance_of(PushSalesforceLead).to receive(:exec).with(hash_including({ newsletter: true }))
-      described_class.call(params: params, contracts_required: true, user: User.last)
+      it 'does NOT sign up user for the newsletter when NOT checked' do
+        expect_any_instance_of(PushSalesforceLead).to receive(:exec).with(hash_including({ newsletter: false }))
+        params[:signup][:newsletter] = false
+        described_class.call(params: params, contracts_required: true, user: User.last)
+      end
     end
 
-    it 'does NOT sign up user for the newsletter when NOT checked' do
-      expect_any_instance_of(PushSalesforceLead).to receive(:exec).with(hash_including({ newsletter: false }))
-      params[:info][:newsletter] = false
-      described_class.call(params: params, contracts_required: true, user: User.last)
+    context 'when user has NO email address' do
+      before do
+        FactoryBot.create(:user, state: 'unverified')
+      end
+
+      it 'creates an email address for the user' do
+        expect {
+          described_class.call(params: params, user: User.last)
+        }.to(
```

Comment:
this is totally subjective, but i think this to() would read a lot better on one line.